pkg/controllers/disruption/controller.go

```diff
@@ -130,7 +130,7 @@ func (c *Controller) Reconcile(ctx context.Context) (reconcile.Result, error) {
 		c.recordRun(fmt.Sprintf("%T", m))
 		success, err := c.disrupt(ctx, m)
 		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("disrupting via %q, %w", m.Type(), err)
+			return reconcile.Result{}, fmt.Errorf("disrupting via %s, %w", string(m.Reason()), err)
```

Comment:
Do you actually need to do the string conversion here? I believe this should just get handled for you by the formatting args.

https://go.dev/play/p/snNO7zCm0sf